### PR TITLE
Allow Pokemon/PvP specific city subscriptions

### PR DESCRIPTION
**Pull request by versx**
Fixes: #64 
Fixes: #67
Fixes: https://github.com/versx/WhMgr-UI/issues/13

- Allows subscriptions for Pokemon and PvP for specific cities instead of relying on assigned Discord roles.
- Automatically disable user subscriptions if not a supporter/donor
- Combine subscription entries instead of separate per city
- Ignore subscribed geofences if distance is set
- Adds clickable distance location link to subscription info command
- Adds subscription amount limits per user per Discord server.
- Send subscriptions if distance or geofence is met.

### CODE REVIEW

**Comment by versx** [comment]
Anyone test? :eyes:

**Comment by versx** [comment]
Anyone test? :eyes:

**Review by Swiss-Rainbow — COMMENTED** [review]
I've been running it for 5 days, it's running fine

**Comment by versx** [comment]
Anyone running this having issues or can I merge?

File: src/Data/Subscriptions/SubscriptionProcessor.cs [hunk 1]
```diff
@@ -167,24 +167,12 @@ public async Task ProcessPokemonSubscription(PokemonData pkmn)
...
-                        }
-                        _logger.Debug($"Distance matches for user {member.DisplayName} ({member.Id}) for Pokemon {pokemon.Name}: {distance}/{user.DistanceM}");
-                    }
-                    else
-                    {
-                        // Otherwise check if in subscribed geofences
-                        if (!subscribedPokemon.Areas.Contains(loc.Name.ToLower()))
-                            continue;
-                    }
+                    var distanceMatches = user.DistanceM > 0 && user.DistanceM > new Coordinates(user.Latitude, user.Longitude).DistanceTo(new Coordinates(pkmn.Latitude, pkmn.Longitude));
```
Comment (versx) [inline src/Data/Subscriptions/SubscriptionProcessor.cs:170]: Also check if `user.DistanceM == 0`

**Pull request merged**